roles/galaxy-route/tasks/main.yml

```diff
@@ -15,6 +15,26 @@
     - ingress_type is defined
     - ('route' == ingress_type|lower) or ('ingress' == ingress_type|lower)
 
+- name: Remove deprecated {{ deployment_type }} Routes for {{ ansible_operator_meta.name }} from {{ ansible_operator_meta.namespace }} namespace
+  kubernetes.core.k8s:
+    state: absent
+    api_version: v1
+    kind: Route
+    namespace: "{{ ansible_operator_meta.namespace }}"
+    name: "{{ item }}"
+  loop:
+    - "{{ ansible_operator_meta.name }}-ansible-pulp-ansible-galaxy"
+    - "{{ ansible_operator_meta.name }}-api-v3"
+    - "{{ ansible_operator_meta.name }}-auth"
+    - "{{ ansible_operator_meta.name }}-container-extensions-v2"
+    - "{{ ansible_operator_meta.name }}-container-pulp-container"
+    - "{{ ansible_operator_meta.name }}-container-token"
+    - "{{ ansible_operator_meta.name }}-container-v2"
+    - "{{ ansible_operator_meta.name }}-content"
+    - "{{ ansible_operator_meta.name }}-galaxy-api"
+    - "{{ ansible_operator_meta.name }}-galaxy-ui"
+    - "{{ ansible_operator_meta.name }}-galaxy-ui2"
```

Comment:
Hi, this task fails in K8s environment with the error `Failed to find exact match for v1.Route by [kind, name, singularName, shortNames]` and causing the deployment to be failed.

Formatted logs from Operator:

```bash
TASK [galaxy-route : Remove deprecated galaxy Routes for galaxy from galaxy namespace] ***
task path: /opt/ansible/roles/galaxy-route/tasks/main.yml:18
...
failed: [localhost] (item=galaxy-ansible-pulp-ansible-galaxy) => {\"ansible_loop_var\": \"item\", \"changed\": false, \"item\": \"galaxy-ansible-pulp-ansible-galaxy\", \"msg\": \"Failed to find exact match for v1.Route by [kind, name, singularName, shortNames]\"}
failed: [localhost] (item=galaxy-api-v3) => {\"ansible_loop_var\": \"item\", \"changed\": false, \"item\": \"galaxy-api-v3\", \"msg\": \"Failed to find exact match for v1.Route by [kind, name, singularName, shortNames]\"}
failed: [localhost] (item=galaxy-auth) => {\"ansible_loop_var\": \"item\", \"changed\": false, \"item\": \"galaxy-auth\", \"msg\": \"Failed to find exact match for v1.Route by [kind, name, singularName, shortNames]\"}
failed: [localhost] (item=galaxy-container-extensions-v2) => {\"ansible_loop_var\": \"item\", \"changed\": false, \"item\": \"galaxy-container-extensions-v2\", \"msg\": \"Failed to find exact match for v1.Route by [kind, name, singularName, shortNames]\"}
failed: [localhost] (item=galaxy-container-pulp-container) => {\"ansible_loop_var\": \"item\", \"changed\": false, \"item\": \"galaxy-container-pulp-container\", \"msg\": \"Failed to find exact match for v1.Route by [kind, name, singularName, shortNames]\"}
failed: [localhost] (item=galaxy-container-token) => {\"ansible_loop_var\": \"item\", \"changed\": false, \"item\": \"galaxy-container-token\", \"msg\": \"Failed to find exact match for v1.Route by [kind, name, singularName, shortNames]\"}
failed: [localhost] (item=galaxy-container-v2) => {\"ansible_loop_var\": \"item\", \"changed\": false, \"item\": \"galaxy-container-v2\", \"msg\": \"Failed to find exact match for v1.Route by [kind, name, singularName, shortNames]\"}
failed: [localhost] (item=galaxy-content) => {\"ansible_loop_var\": \"item\", \"changed\": false, \"item\": \"galaxy-content\", \"msg\": \"Failed to find exact match for v1.Route by [kind, name, singularName, shortNames]\"}
failed: [localhost] (item=galaxy-galaxy-api) => {\"ansible_loop_var\": \"item\", \"changed\": false, \"item\": \"galaxy-galaxy-api\", \"msg\": \"Failed to find exact match for v1.Route by [kind, name, singularName, shortNames]\"}
failed: [localhost] (item=galaxy-galaxy-ui) => {\"ansible_loop_var\": \"item\", \"changed\": false, \"item\": \"galaxy-galaxy-ui\", \"msg\": \"Failed to find exact match for v1.Route by [kind, name, singularName, shortNames]\"}
failed: [localhost] (item=galaxy-galaxy-ui2) => {\"ansible_loop_var\": \"item\", \"changed\": false, \"item\": \"galaxy-galaxy-ui2\", \"msg\": \"Failed to find exact match for v1.Route by [kind, name, singularName, shortNames]\"}

PLAY RECAP *********************************************************************
localhost                  : ok=114  changed=46   unreachable=0    failed=1   skipped=73   rescued=0    ignored=1   
```

Could you please make this task triggered only in OpenShift?

```suggestion
    - "{{ ansible_operator_meta.name }}-galaxy-ui2"
  when: is_openshift
```

Reply:
Thanks, I didn't realize we were reusing the same role for non-openshift (the Routes naming threw me off).

Reply:
Thanks for the fix!

P.S. Interestingly, the role name is route, but the template file name is ingress😃

Reply:
Yea.. we were just discussing the same thing. 😄  I plan to do more cleanup and renaming of roles in the future so I'll fix it during that effort.